README
Lavender Throne is released under the GNU General Public license, version 3.  
See file COPYING for details.

Lavender Throne makes use of the PCG random number generator.  Source code, 
found in the files src/pcg_basic.c and src/pcg_basic.h, is released under the 
Apache License version 2.0 and was written by Melissa O'Neill.  See 
http://www.apache.org/licenses/LICENSE-2.0 for details.

You can turn on debugging code (which gives some shortcuts during gameplay, 
e.g. hidden doors are marked and creatures can be killed with a single hit) by 
uncommenting the line with "#define _D_DEBUG" in those files where it exists 
(currently leveldisp.c and attack.c).  For some reason, defining it at compile 
time with "make CFLAGS=-D_D_DEBUG" causes placeDoors to segfault.  No clue why.

HOW TO PLAY

Move with numeric keypad, in cardinal directions and diagonally.  Numeric keypad '.' uses a move in place.

Map symbols:
.	open floor
#	wall
<	up stair
>	down stair
+	closed door
'	open door

Commands:
c	close an open door
d	drop an item
e	eat food
i	view inventory list
n	name an item
o	open a closed door
p	pick fruit from a plant
r	remove a piece of armor
s	search for hidden doors
S	save game
t	throw an object
u	unwield currently-wielded item
w	wield an object (anything can be wielded, including things that aren't explicitly weapons)
W	wear a piece of armor
;	look at an object on the map (navigate the cursor with the movement 
	keys, then hit enter/return)
#	un-name an item
,	pick up an item
<	go up an up stair
>	go down a down stair

ChangeLog
for version 0.12
  * code refactoring: alphabetized functions in input.c and moved declarations to input.h (kmw, 2017-12-26)
  * when a non-player creature kills another creature, that creature is properly killed in-game (i.e. via killCreature()) rather than just
    ignored (kmw, 2018-04-28)
  * code refactoring: alphabetized functions in game.c and moved declarations to game.h (kmw, 2017-12-26)

for version 0.11
  * release (kmw, 2017-12-26)
  * code refactoring: alphabetized functions in flooritems.c and moved declarations to flooritems.h
    (kmw, 2017-12-26)
  * clear "...(space to cancel)" message when canceling actions that use inventory window (kmw, 2017-12-24)
  * code refactoring: alphabetized functions in faction.c and moved declarations to faction.h
    (kmw, 2017-12-24)
  * attempting to wield a worn item now fails (kmw, 2017-12-23)
  * code refactoring: alphabetized functions in eat.c and moved declarations to eat.h (kmw, 2017-12-23)
  * attempting to wear a wielded item now fails (kmw, 2017-12-23)
  * code refactoring: alphabetized functions in creaturemove.c and moved declarations to creaturemgmt.h
    (kmw, 2017-12-23)
  * Display confirmation message after save complete (kmw, 2017-12-23)
  * code refactoring: alphabetized functions in creaturemgmt.c and moved declarations to creaturemgmt.h
    (kmw, 2017-12-23)
  * fixed bug that let you drop a worn item (kmw, 2017-12-22)
  * code refactoring: alphabetized functions in creaturelist.c and moved declarations to creaturelist.h
    (kmw, 2017-12-22)
  * creatures only display if they're within visual range (kmw, 2017-12-22)
  * code refactoring: alphabetized functions in creatureinventory.c and moved declarations to
    creatureinventory.h (kmw, 2017-09-14)
  * it is now possible to use a move staying in place (kmw, 2017-09-14)
  * code refactoring: alphabetized functions in creatureinit.c and moved declarations to creatureinit.h
    (kmw, 2017-09-14)
  * creatures will attack and fight back if attacked (kmw, 2017-09-14)
  * code refactoring: alphabetized functions in creaturegen.c and moved declarations to creaturegen.h
    (kmw, 2017-09-13)
  * creatures will follow their targets, with a probability based on their aggressiveness (kmw, 2017-09-13)
  * code refactoring: moved creatureeat.c function declarations to creatureeat.h (kmw, 2017-09-12)
  * creatures will now choose an enemy creature to target based on distance (kmw, 2017-09-12)
  * code refactoring: moved colors.c function declarations to colors.h (kmw, 2017-09-11)

for version 0.10
  * release (kmw, 2017-09-11)
  * code refactoring: alphabetized functions in chargen.c and moved declarations to chargen.h
    (kmw, 2017-09-09)
  * creatures will wear armor if it's better than what they have (kmw, 2017-09-09)
  * code refactoring: alphabetized functions in attack.c and moved declarations to attack.h
    (kmw, 2017-09-09)
  * creatures now pick up armor if it's better than what they have for a given slot (kmw, 2017-09-09)
  * code refactoring: moved function declarations for allocator.c to new header allocator.h
    (kmw, 2017-09-08)
  * creatures do not wield picked-up weapons until next turn (kmw, 2017-09-08)
  * code refactoring: alphabetized functions in allocator.c (kmw, 2017-07-07)
  * looking at creatures tells you about their weapon if they're wielding one (kmw, 2017-07-07)
  * creatures will pick up weapons if they're better than current or backup weapon (kmw, 2017-07-07)
  * hungry creatures will now pick up food if they randomly happen to pass by a food source, though
    they do not actively seek food out (kmw, 2017-06-26)

for version 0.9
  * release (kmw, 2017-05-21)
  * unowned items are saved and restored (kmw, 2017-05-21)
  * plants are saved and restored (kmw, 2017-04-24)
  * seeds are updated after being restored from save (kmw, 2017-04-24)
  * save and restore dungeon map and creatures (kmw, 2017-04-24)
  * refactoring object-storage code in save.c (kmw, 2017-04-23)
  * global status save and restore (kmw, 2017-04-18)
  * move global status initialization into separate function to facilitate save restore (kmw, 2017-04-18)
  * restore player data from savefile (kmw, 2017-04-18)
  * serialize and store creature, item, and strings when saving game (kmw, 2017-04-17)

for version 0.8
  * release (kmw, 2017-04-10)
  * seeds that lay in place will plant and grow (kmw, 2017-04-10)
  * seeds that are not planted in time will rot away (kmw, 2017-04-09)
  * plants are updated every turn rather than every move (kmw, 2017-04-09)
  * addContents() in map.c also now sets the location in item struct (kmw, 2017-04-09)
  * eating a berry creates a seed (kmw, 2017-04-09)
  * fixed bug relating to freeing item type names of fruit (kmw, 2017-03-31)
  * fruit and corpses can be eaten (kmw, 2017-03-31)
  * fixed bug that caused segfault when scrolling past end of full inventory list display (kmw, 2017-03-31)
  * fruit can now be picked from plants (kmw, 2017-03-31)
  * non-player creature life cycle and hunger update every turn instead of every move (kmw, 2017-03-30)
  * it is now possible to die of hunger (kmw, 2017-03-30)
  * plants will die if they are trampled enough (kmw, 2017-03-30)
  * added some new plant species (kmw, 2017-03-28)
  * plant growth and production updates every turn (kmw, 2017-03-27)
  * plants bearing fruit are highlighted on-screen (kmw, 2017-03-27)
  * look command now works on plants (kmw, 2017-03-27)
  * generate and place plants (kmw, 2017-02-26)
  * update life cycle for non-player creatures (kmw, 2017-02-16)

for version 0.7
  * release (kmw, 2017-02-15)
  * thrown objects now do damage when they hit a creature (kmw, 2017-02-15)
  * set weight of corpses when they are spawned after creature is killed (kmw, 2017-02-15)
  * refactored attack code to make different types of attacks easier to add (kmw, 2017-02-15)
  * objects can now be thrown (kmw, 2017-02-13)
  * added shields (kmw, 2017-02-06)

src/item.c
/*  This file is part of Lavender Throne.
 *  Copyright 2016 by Kurt Weber
 *
 *  Lavender Throne is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  Lavender Throne is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with Lavender Throne.  If not, see <http://www.gnu.org/licenses/>.
 */

#define _ITEM_C

#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include "colors.h"
#include "item.h"
#include "lvt.h"

itemType *itemTypes[ITEM_TYPE_MAX];

void initItems(){
	initCorpses();
	initLongswords();
	initAxes();
	initDaggers();
	initShirts();
	initUnderarmor();
	initShields();
	initFruit();
	initSeeds();
	
	return;
}

void initShields(){
	shieldSubClassId i;
	char dispChar = '[';
	
	itemTypes[ITEM_TYPE_SHIELD] = calloc(ITEM_SHIELD_MAX, sizeof(itemType));
	
	// wooden shield
	{
		initItemType(&itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_WOOD]);
		itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_WOOD].baseDamage = 0;
		itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_WOOD].baseToHit = 0;
		itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_WOOD].baseArmor = 3;
		itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_WOOD].throwable = false;
		itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_WOOD].twoHandedWield = false;
		itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_WOOD].fragile = false;
		itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_WOOD].itemName = calloc(14, sizeof(char));
		strcpy(itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_WOOD].itemName, "wooden shield");
		itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_WOOD].dispChar = dispChar;
		itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_WOOD].color = BrownBlack;
		itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_WOOD].attrs = 0;
		itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_WOOD].weight = 7;
		itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_WOOD].possibleSlots = ARMOR_SHIELD;
	}
	
	// steel shield
	{
		initItemType(&itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_STEEL]);
		itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_STEEL].baseDamage = 0;
		itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_STEEL].baseToHit = 0;
		itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_STEEL].baseArmor = 5;
		itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_STEEL].throwable = false;
		itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_STEEL].twoHandedWield = false;
		itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_STEEL].fragile = false;
		itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_STEEL].itemName = calloc(13, sizeof(char));
		strcpy(itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_STEEL].itemName, "steel shield");
		itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_STEEL].dispChar = dispChar;
		itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_STEEL].color = WhiteBlack;
		itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_STEEL].attrs = 0;
		itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_STEEL].weight = 10;
		itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_STEEL].possibleSlots = ARMOR_SHIELD;
	}
	
	// silver shield
	{
		initItemType(&itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_SILVER]);
		itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_SILVER].baseDamage = 0;
		itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_SILVER].baseToHit = 0;
		itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_SILVER].baseArmor = 6;
		itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_SILVER].throwable = false;
		itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_SILVER].twoHandedWield = false;
		itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_SILVER].fragile = false;
		itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_SILVER].itemName = calloc(14, sizeof(char));
		strcpy(itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_SILVER].itemName, "silver shield");
		itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_SILVER].dispChar = dispChar;
		itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_SILVER].color = WhiteBlack;
		itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_SILVER].attrs = A_BOLD;
		itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_SILVER].weight = 15;
		itemTypes[ITEM_TYPE_SHIELD][ITEM_SHIELD_SILVER].possibleSlots = ARMOR_SHIELD;
	}
	return;
}

void initUnderarmor(){
	char dispChar = ']';
	
	itemTypes[ITEM_TYPE_UNDERARMOR] = calloc(ITEM_UNDERARMOR_MAX, sizeof(itemType));
	
	// underarmor
	{
		initItemType(&itemTypes[ITEM_TYPE_UNDERARMOR][ITEM_UNDERARMOR_CHAINMAIL]);
		itemTypes[ITEM_TYPE_UNDERARMOR][ITEM_UNDERARMOR_CHAINMAIL].baseDamage = 0;
		itemTypes[ITEM_TYPE_UNDERARMOR][ITEM_UNDERARMOR_CHAINMAIL].baseToHit = 0;
		itemTypes[ITEM_TYPE_UNDERARMOR][ITEM_UNDERARMOR_CHAINMAIL].baseArmor = 5;
		itemTypes[ITEM_TYPE_UNDERARMOR][ITEM_UNDERARMOR_CHAINMAIL].throwable = false;
		itemTypes[ITEM_TYPE_UNDERARMOR][ITEM_UNDERARMOR_CHAINMAIL].twoHandedWield = false;
		itemTypes[ITEM_TYPE_UNDERARMOR][ITEM_UNDERARMOR_CHAINMAIL].fragile = false;
		itemTypes[ITEM_TYPE_UNDERARMOR][ITEM_UNDERARMOR_CHAINMAIL].itemName = calloc(11, sizeof(char));
		strcpy(itemTypes[ITEM_TYPE_UNDERARMOR][ITEM_UNDERARMOR_CHAINMAIL].itemName, "chain mail");
		itemTypes[ITEM_TYPE_UNDERARMOR][ITEM_UNDERARMOR_CHAINMAIL].dispChar = dispChar;
		itemTypes[ITEM_TYPE_UNDERARMOR][ITEM_UNDERARMOR_CHAINMAIL].color = WhiteBlack;
		itemTypes[ITEM_TYPE_UNDERARMOR][ITEM_UNDERARMOR_CHAINMAIL].attrs = A_BOLD;
		itemTypes[ITEM_TYPE_UNDERARMOR][ITEM_UNDERARMOR_CHAINMAIL].weight = 10;
		itemTypes[ITEM_TYPE_UNDERARMOR][ITEM_UNDERARMOR_CHAINMAIL].possibleSlots = ARMOR_UNDERARMOR;
	}
	
	return;
}

void initShirts(){
	char dispChar = ']';
	
	itemTypes[ITEM_TYPE_SHIRT] = calloc(ITEM_SHIRT_MAX, sizeof(itemType));
	
	// t-shirt
	{
		initItemType(&itemTypes[ITEM_TYPE_SHIRT][ITEM_SHIRT_HAWAIIAN]);
		itemTypes[ITEM_TYPE_SHIRT][ITEM_SHIRT_TSHIRT].baseDamage = 0;
		itemTypes[ITEM_TYPE_SHIRT][ITEM_SHIRT_TSHIRT].baseToHit = 0;
		itemTypes[ITEM_TYPE_SHIRT][ITEM_SHIRT_TSHIRT].baseArmor = 1;
		itemTypes[ITEM_TYPE_SHIRT][ITEM_SHIRT_TSHIRT].throwable = false;
		itemTypes[ITEM_TYPE_SHIRT][ITEM_SHIRT_TSHIRT].twoHandedWield = false;
		itemTypes[ITEM_TYPE_SHIRT][ITEM_SHIRT_TSHIRT].fragile = false;
		itemTypes[ITEM_TYPE_SHIRT][ITEM_SHIRT_TSHIRT].itemName = calloc(8, sizeof(char));
		strcpy(itemTypes[ITEM_TYPE_SHIRT][ITEM_SHIRT_TSHIRT].itemName, "t-shirt");
		itemTypes[ITEM_TYPE_SHIRT][ITEM_SHIRT_TSHIRT].dispChar = dispChar;
		itemTypes[ITEM_TYPE_SHIRT][ITEM_SHIRT_TSHIRT].color = WhiteBlack;
		itemTypes[ITEM_TYPE_SHIRT][ITEM_SHIRT_TSHIRT].attrs = 0;
		itemTypes[ITEM_TYPE_SHIRT][ITEM_SHIRT_TSHIRT].weight = 1;
		itemTypes[ITEM_TYPE_SHIRT][ITEM_SHIRT_TSHIRT].possibleSlots = ARMOR_SHIRT;
	}
	
	// Hawaiian shirt
	{
		initItemType(&itemTypes[ITEM_TYPE_SHIRT][ITEM_SHIRT_HAWAIIAN]);
		itemTypes[ITEM_TYPE_SHIRT][ITEM_SHIRT_HAWAIIAN].baseDamage = 0;
		itemTypes[ITEM_TYPE_SHIRT][ITEM_SHIRT_HAWAIIAN].baseToHit = 0;
		itemTypes[ITEM_TYPE_SHIRT][ITEM_SHIRT_HAWAIIAN].baseArmor = 1;
		itemTypes[ITEM_TYPE_SHIRT][ITEM_SHIRT_HAWAIIAN].throwable = false;
		itemTypes[ITEM_TYPE_SHIRT][ITEM_SHIRT_HAWAIIAN].twoHandedWield = false;
		itemTypes[ITEM_TYPE_SHIRT][ITEM_SHIRT_HAWAIIAN].fragile = false;
		itemTypes[ITEM_TYPE_SHIRT][ITEM_SHIRT_HAWAIIAN].itemName = calloc(15, sizeof(char));
		strcpy(itemTypes[ITEM_TYPE_SHIRT][ITEM_SHIRT_HAWAIIAN].itemName, "Hawaiian shirt");
		itemTypes[ITEM_TYPE_SHIRT][ITEM_SHIRT_HAWAIIAN].dispChar = dispChar;
		itemTypes[ITEM_TYPE_SHIRT][ITEM_SHIRT_HAWAIIAN].color = MagentaBlack;
		itemTypes[ITEM_TYPE_SHIRT][ITEM_SHIRT_HAWAIIAN].attrs = 0;
		itemTypes[ITEM_TYPE_SHIRT][ITEM_SHIRT_HAWAIIAN].weight = 1;
		itemTypes[ITEM_TYPE_SHIRT][ITEM_SHIRT_HAWAIIAN].possibleSlots = ARMOR_SHIRT;
	}
	
	return;
}

void initDaggers(){
	char dispChar = '/';
	
	itemTypes[ITEM_TYPE_DAGGER] = calloc(ITEM_DAGGER_MAX, sizeof(itemType));
	
	// stiletto dagger
	{
		initItemType(&itemTypes[ITEM_TYPE_DAGGER][ITEM_DAGGER_STILETTO]);
		itemTypes[ITEM_TYPE_DAGGER][ITEM_DAGGER_STILETTO].baseDamage = 2;
		itemTypes[ITEM_TYPE_DAGGER][ITEM_DAGGER_STILETTO].baseToHit = 20;
		itemTypes[ITEM_TYPE_DAGGER][ITEM_DAGGER_STILETTO].baseArmor = 0;
		itemTypes[ITEM_TYPE_DAGGER][ITEM_DAGGER_STILETTO].throwable = true;
		itemTypes[ITEM_TYPE_DAGGER][ITEM_DAGGER_STILETTO].twoHandedWield = false;
		itemTypes[ITEM_TYPE_DAGGER][ITEM_DAGGER_STILETTO].fragile = false;
		itemTypes[ITEM_TYPE_DAGGER][ITEM_DAGGER_STILETTO].itemName = calloc(16, sizeof(char));
		strcpy(itemTypes[ITEM_TYPE_DAGGER][ITEM_DAGGER_STILETTO].itemName, "stiletto dagger");
		itemTypes[ITEM_TYPE_DAGGER][ITEM_DAGGER_STILETTO].dispChar = dispChar;
		itemTypes[ITEM_TYPE_DAGGER][ITEM_DAGGER_STILETTO].color = BlueBlack;
		itemTypes[ITEM_TYPE_DAGGER][ITEM_DAGGER_STILETTO].attrs = 0;
		itemTypes[ITEM_TYPE_DAGGER][ITEM_DAGGER_STILETTO].weight = 1;
		itemTypes[ITEM_TYPE_DAGGER][ITEM_DAGGER_STILETTO].possibleSlots = 0;
	}
	
	// baselard dagger
	{
		initItemType(&itemTypes[ITEM_TYPE_DAGGER][ITEM_DAGGER_BASELARD]);
		itemTypes[ITEM_TYPE_DAGGER][ITEM_DAGGER_BASELARD].baseDamage = 4;
		itemTypes[ITEM_TYPE_DAGGER][ITEM_DAGGER_BASELARD].baseToHit = 15;
		itemTypes[ITEM_TYPE_DAGGER][ITEM_DAGGER_BASELARD].baseArmor = 0;
		itemTypes[ITEM_TYPE_DAGGER][ITEM_DAGGER_BASELARD].throwable = true;
		itemTypes[ITEM_TYPE_DAGGER][ITEM_DAGGER_BASELARD].twoHandedWield = false;
		itemTypes[ITEM_TYPE_DAGGER][ITEM_DAGGER_BASELARD].fragile = false;
		itemTypes[ITEM_TYPE_DAGGER][ITEM_DAGGER_BASELARD].itemName = calloc(16, sizeof(char));
		strcpy(itemTypes[ITEM_TYPE_DAGGER][ITEM_DAGGER_BASELARD].itemName, "baselard dagger");
		itemTypes[ITEM_TYPE_DAGGER][ITEM_DAGGER_BASELARD].dispChar = dispChar;
		itemTypes[ITEM_TYPE_DAGGER][ITEM_DAGGER_BASELARD].color = BlueBlack;
		itemTypes[ITEM_TYPE_DAGGER][ITEM_DAGGER_BASELARD].attrs = 0;
		itemTypes[ITEM_TYPE_DAGGER][ITEM_DAGGER_BASELARD].weight = 1;
		itemTypes[ITEM_TYPE_DAGGER][ITEM_DAGGER_BASELARD].possibleSlots = 0;
	}
	
	return;
}

void initAxes(){
	char dispChar = '/';
	
	itemTypes[ITEM_TYPE_AXE] = calloc(ITEM_AXE_MAX, sizeof(itemType));
	
	// throwing axe
	{
		initItemType(&itemTypes[ITEM_TYPE_AXE][ITEM_AXE_THROWING]);
		itemTypes[ITEM_TYPE_AXE][ITEM_AXE_THROWING].baseDamage = 3;
		itemTypes[ITEM_TYPE_AXE][ITEM_AXE_THROWING].baseToHit = 7;
		itemTypes[ITEM_TYPE_AXE][ITEM_AXE_THROWING].baseArmor = 0;
		itemTypes[ITEM_TYPE_AXE][ITEM_AXE_THROWING].throwable = true;
		itemTypes[ITEM_TYPE_AXE][ITEM_AXE_THROWING].twoHandedWield = false;
		itemTypes[ITEM_TYPE_AXE][ITEM_AXE_THROWING].fragile = false;
		itemTypes[ITEM_TYPE_AXE][ITEM_AXE_THROWING].itemName = calloc(13, sizeof(char));
		strcpy(itemTypes[ITEM_TYPE_AXE][ITEM_AXE_THROWING].itemName, "throwing axe");
		itemTypes[ITEM_TYPE_AXE][ITEM_AXE_THROWING].dispChar = dispChar;
		itemTypes[ITEM_TYPE_AXE][ITEM_AXE_THROWING].color = BrownBlack;
		itemTypes[ITEM_TYPE_AXE][ITEM_AXE_THROWING].attrs = 0;
		itemTypes[ITEM_TYPE_AXE][ITEM_AXE_THROWING].weight = 2;
		itemTypes[ITEM_TYPE_AXE][ITEM_AXE_THROWING].possibleSlots = 0;
	}
	
	// battle axe
	{
		initItemType(&itemTypes[ITEM_TYPE_AXE][ITEM_AXE_BATTLE]);
		itemTypes[ITEM_TYPE_AXE][ITEM_AXE_BATTLE].baseDamage = 9;
		itemTypes[ITEM_TYPE_AXE][ITEM_AXE_BATTLE].baseToHit = 5;
		itemTypes[ITEM_TYPE_AXE][ITEM_AXE_BATTLE].baseArmor = 0;
		itemTypes[ITEM_TYPE_AXE][ITEM_AXE_BATTLE].throwable = true;
		itemTypes[ITEM_TYPE_AXE][ITEM_AXE_BATTLE].twoHandedWield = true;
		itemTypes[ITEM_TYPE_AXE][ITEM_AXE_BATTLE].fragile = false;
		itemTypes[ITEM_TYPE_AXE][ITEM_AXE_BATTLE].itemName = calloc(11, sizeof(char));
		strcpy(itemTypes[ITEM_TYPE_AXE][ITEM_AXE_BATTLE].itemName, "battle axe");
		itemTypes[ITEM_TYPE_AXE][ITEM_AXE_BATTLE].dispChar = dispChar;
		itemTypes[ITEM_TYPE_AXE][ITEM_AXE_BATTLE].color = BrownBlack;
		itemTypes[ITEM_TYPE_AXE][ITEM_AXE_BATTLE].attrs = 0;
		itemTypes[ITEM_TYPE_AXE][ITEM_AXE_BATTLE].weight = 2;
		itemTypes[ITEM_TYPE_AXE][ITEM_AXE_THROWING].possibleSlots = 0;
	}
	
	return;
}

void initLongswords(){
	char dispChar = '/';
	
	itemTypes[ITEM_TYPE_LONGSWORD] = calloc(ITEM_LONGSWORD_MAX, sizeof(itemType));
	
	// steel longsword
	{
		initItemType(&itemTypes[ITEM_TYPE_LONGSWORD][ITEM_LONGSWORD_STEEL]);
		itemTypes[ITEM_TYPE_LONGSWORD][ITEM_LONGSWORD_STEEL].baseDamage = 5;
		itemTypes[ITEM_TYPE_LONGSWORD][ITEM_LONGSWORD_STEEL].baseToHit = 4;
		itemTypes[ITEM_TYPE_LONGSWORD][ITEM_LONGSWORD_STEEL].baseArmor = 0;
		itemTypes[ITEM_TYPE_LONGSWORD][ITEM_LONGSWORD_STEEL].throwable = false;
		itemTypes[ITEM_TYPE_LONGSWORD][ITEM_LONGSWORD_STEEL].twoHandedWield = false;
		itemTypes[ITEM_TYPE_LONGSWORD][ITEM_LONGSWORD_STEEL].fragile = false;
		itemTypes[ITEM_TYPE_LONGSWORD][ITEM_LONGSWORD_STEEL].itemName = calloc(16, sizeof(char));
		strcpy(itemTypes[ITEM_TYPE_LONGSWORD][ITEM_LONGSWORD_STEEL].itemName, "steel longsword");
		itemTypes[ITEM_TYPE_LONGSWORD][ITEM_LONGSWORD_STEEL].dispChar = dispChar;
		itemTypes[ITEM_TYPE_LONGSWORD][ITEM_LONGSWORD_STEEL].color = WhiteBlack;
		itemTypes[ITEM_TYPE_LONGSWORD][ITEM_LONGSWORD_STEEL].attrs = 0;
		itemTypes[ITEM_TYPE_LONGSWORD][ITEM_LONGSWORD_STEEL].weight = 8;
		itemTypes[ITEM_TYPE_LONGSWORD][ITEM_LONGSWORD_STEEL].possibleSlots = 0;
	}
	
	// I use braces like this because it lets me code-fold these blocks out of the way in my text editor
	
	// silver longsword
	{
		initItemType(&itemTypes[ITEM_TYPE_LONGSWORD][ITEM_LONGSWORD_SILVER]);
		itemTypes[ITEM_TYPE_LONGSWORD][ITEM_LONGSWORD_SILVER].baseDamage = 7;
		itemTypes[ITEM_TYPE_LONGSWORD][ITEM_LONGSWORD_SILVER].baseToHit = 3;
		itemTypes[ITEM_TYPE_LONGSWORD][ITEM_LONGSWORD_SILVER].baseArmor = 0;
		itemTypes[ITEM_TYPE_LONGSWORD][ITEM_LONGSWORD_SILVER].throwable = false;
		itemTypes[ITEM_TYPE_LONGSWORD][ITEM_LONGSWORD_SILVER].twoHandedWield = false;
		itemTypes[ITEM_TYPE_LONGSWORD][ITEM_LONGSWORD_SILVER].fragile = false;
		itemTypes[ITEM_TYPE_LONGSWORD][ITEM_LONGSWORD_SILVER].itemName = calloc(17, sizeof(char));
		strcpy(itemTypes[ITEM_TYPE_LONGSWORD][ITEM_LONGSWORD_SILVER].itemName, "silver longsword");
		itemTypes[ITEM_TYPE_LONGSWORD][ITEM_LONGSWORD_SILVER].dispChar = dispChar;
		itemTypes[ITEM_TYPE_LONGSWORD][ITEM_LONGSWORD_SILVER].color = CyanBlack;
		itemTypes[ITEM_TYPE_LONGSWORD][ITEM_LONGSWORD_SILVER].attrs = 0;
		itemTypes[ITEM_TYPE_LONGSWORD][ITEM_LONGSWORD_SILVER].weight = 10;
		itemTypes[ITEM_TYPE_LONGSWORD][ITEM_LONGSWORD_SILVER].possibleSlots = 0;
	}
	
	return;
}

void initSeeds(){
	seedSubClassId i;
	char *seedSubClassNames[] = {"blueberry", "blackberry", "strawberry", "raspberry"};
	colorPairs seedColors[] = {CyanBlack, BlueBlack, RedBlack, MagentaBlack};
	
	itemTypes[ITEM_TYPE_SEED] = calloc(ITEM_SEED_MAX, sizeof(itemType));
	
	for (i = 0; i < ITEM_SEED_MAX; i++){
		initItemType(&itemTypes[ITEM_TYPE_SEED][i]);
		itemTypes[ITEM_TYPE_SEED][i].itemName = calloc(strlen(seedSubClassNames[i]) + 5 + 1, sizeof(char));
		strcpy(itemTypes[ITEM_TYPE_SEED][i].itemName, seedSubClassNames[i]);
		strcat(itemTypes[ITEM_TYPE_SEED][i].itemName, " seed");
		itemTypes[ITEM_TYPE_SEED][i].dispChar = '*';
		itemTypes[ITEM_TYPE_SEED][i].color = seedColors[i];
		itemTypes[ITEM_TYPE_SEED][i].attrs = 0;
		itemTypes[ITEM_TYPE_SEED][i].possibleSlots = 0;
		itemTypes[ITEM_TYPE_SEED][i].weight = 1;
	}
	
	return;
}

void initFruit(){
	fruitSubClassId i;
	char *fruitSubClassNames[] = {"blueberry", "blackberry", "strawberry", "raspberry"};
	colorPairs fruitColors[] = {CyanBlack, BlueBlack, RedBlack, MagentaBlack};
	
	itemTypes[ITEM_TYPE_FRUIT] = calloc(ITEM_FRUIT_MAX, sizeof(itemType));
	
	for (i = 0; i < ITEM_FRUIT_MAX; i++){
		initItemType(&itemTypes[ITEM_TYPE_FRUIT][i]);
		itemTypes[ITEM_TYPE_FRUIT][i].itemName = calloc(strlen(fruitSubClassNames[i]) + 1, sizeof(char));
		strcpy(itemTypes[ITEM_TYPE_FRUIT][i].itemName, fruitSubClassNames[i]);
		itemTypes[ITEM_TYPE_FRUIT][i].dispChar = '*';
		itemTypes[ITEM_TYPE_FRUIT][i].color = fruitColors[i];
		itemTypes[ITEM_TYPE_FRUIT][i].attrs = A_BOLD;
		itemTypes[ITEM_TYPE_FRUIT][i].possibleSlots = 0;
		itemTypes[ITEM_TYPE_FRUIT][i].weight = 2;
	}
	
	return;
}

void initCorpses(){
	corpseSubClassId i;
	char *corpseSubClassNames[] = {"human", "halfling", "elf", "dwarf", "bear"};
	colorPairs corpseColors[] = {WhiteBlack, BlueBlack, GreenBlack, RedBlack, MagentaBlack};
	
	itemTypes[ITEM_TYPE_CORPSE] = calloc(ITEM_CORPSE_MAX, sizeof(itemType));
	
	for (i = 0; i < ITEM_CORPSE_MAX; i++){
		initItemType(&itemTypes[ITEM_TYPE_CORPSE][i]);
		itemTypes[ITEM_TYPE_CORPSE][i].baseDamage = 0;
		itemTypes[ITEM_TYPE_CORPSE][i].baseArmor = 0;
		itemTypes[ITEM_TYPE_CORPSE][i].baseToHit = 0;
		itemTypes[ITEM_TYPE_CORPSE][i].throwable = false;
		itemTypes[ITEM_TYPE_CORPSE][i].twoHandedWield = false;
		itemTypes[ITEM_TYPE_CORPSE][i].fragile = false;
		itemTypes[ITEM_TYPE_CORPSE][i].itemName = calloc(strlen(corpseSubClassNames[i]) + 8, sizeof(char));
		strcpy(itemTypes[ITEM_TYPE_CORPSE][i].itemName, corpseSubClassNames[i]);
		strcat(itemTypes[ITEM_TYPE_CORPSE][i].itemName, " corpse");
		itemTypes[ITEM_TYPE_CORPSE][i].dispChar = '%';
		itemTypes[ITEM_TYPE_CORPSE][i].color = corpseColors[i];
		itemTypes[ITEM_TYPE_CORPSE][i].attrs = 0;
		itemTypes[ITEM_TYPE_CORPSE][i].possibleSlots = 0;
	}
	
	return;
}

void initItemType(itemType *in){
	in->itemName = 0;
	in->baseDamage = 0;
	in->baseToHit = 0;
	in->baseArmor = 0;
	in->weight = 0;
	in->throwable = false;
	in->twoHandedWield = false;
	in->fragile = false;
	in->possibleSlots = 0;
	in->color = 0;
	in->dispChar = 0;
	in->attrs = 0;
	return;
}